clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/deploy/ops/loadbalancer/DeleteGoogleInternalLoadBalancerAtomicOperation.groovy

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright 2016 Google, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.clouddriver.google.deploy.ops.loadbalancer
+
+import com.google.api.services.compute.model.BackendService
+import com.google.api.services.compute.model.ForwardingRule
+import com.google.api.services.compute.model.Operation
+import com.google.common.annotations.VisibleForTesting
+import com.netflix.spinnaker.clouddriver.data.task.Task
+import com.netflix.spinnaker.clouddriver.data.task.TaskRepository
+import com.netflix.spinnaker.clouddriver.google.deploy.GCEUtil
+import com.netflix.spinnaker.clouddriver.google.deploy.GoogleOperationPoller
+import com.netflix.spinnaker.clouddriver.google.deploy.description.DeleteGoogleLoadBalancerDescription
+import com.netflix.spinnaker.clouddriver.google.model.callbacks.Utils
+import com.netflix.spinnaker.clouddriver.orchestration.AtomicOperation
+import groovy.util.logging.Slf4j
+import org.springframework.beans.factory.annotation.Autowired
+
+@Slf4j
+class DeleteGoogleInternalLoadBalancerAtomicOperation implements AtomicOperation<Void> {
+  private static final String BASE_PHASE = "DELETE_INTERNAL_LOAD_BALANCER"
+
+  private static Task getTask() {
+    TaskRepository.threadLocalTask.get()
+  }
+
+  @Autowired
+  private GoogleOperationPoller googleOperationPoller
+
+  private DeleteGoogleLoadBalancerDescription description
+
+  @VisibleForTesting
+  GoogleOperationPoller.ThreadSleeper threadSleeper = new GoogleOperationPoller.ThreadSleeper()
+
+  DeleteGoogleInternalLoadBalancerAtomicOperation() {}
+
+  DeleteGoogleInternalLoadBalancerAtomicOperation(DeleteGoogleLoadBalancerDescription description) {
+    this.description = description
+  }
+
+  /**
+   * curl -X POST -H "Content-Type: application/json" -d '[ { "deleteLoadBalancer": { "region": "us-central1", "credentials": "my-account-name", "loadBalancerName": "testlb", "loadBalancerType": "INTERNAL"}} ]' localhost:7002/gce/ops
+   */
+  @Override
+  Void operate(List priorOutputs) {
```

Comment:
It looks like if this operation fails halfway through (deletes some pieces, but fails to delete others), it will be impossible to delete the dangling components on a retry.